### ISSUE TITLE
Add date / time tokens in Toml Module 

### ISSUE THREAD

**Issue by ankingcodes** [chapel-lang/chapel#15364]
This came up in developement of #15228. 
Inline comments in a TOML module were treated as newline/ fullline comments due to which tests of date / time which had inline comments got passed. eg : 
```
 timestamp3 = 06:30:30.1234567 # rounding up
```
was treated as 
```
 timestamp3 = 06:30:30.1234567 
# rounding up
```
and the token for the timestamp would be ` (06:30:30.1234567)`. 
But with the merge of #15228 , 
inline comment would not be considered as fullline and the token would be `(06:30:30.1234567 # rounding up)` which is an invalid token. This occurs because it has not been specified to separate date or time tokens in a line.

**Comment by ankingcodes** [comment]
@ben-albrecht I'll take this one ! 👍

**Comment by ankingcodes** [comment]
Closing this as the feature has been implemented by #15228
